src/Rector/MethodCall/ResponseHelperCallToJsonResponseRector.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RectorLaravel\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Name\FullyQualified;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://laravel.com/docs/7.x/database-testing#creating-models
+ * @changelog https://laravel.com/docs/8.x/database-testing#instantiating-models
+ *
+ * @see \RectorLaravel\Tests\Rector\FuncCall\FactoryFuncCallToStaticCallRector\FactoryFuncCallToStaticCallRectorTest
+ */
+final class ResponseHelperCallToJsonResponseRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Use new JsonResponse instead of response()->json()', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+response()->json(['key' => 'value']);
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+return new JsonResponse(['key' => 'value']);
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param  MethodCall  $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isName($node->name, 'json')) {
+            return null;
+        }
+
+        if (! $this->isName($node->var, 'response')) {
```

Comment:
This check is not sufficient, since it could work with cases like `$response->json()` when it's not an instance of `JsonResponse`. We should instead make this check for the `ObjectType`.

Reply:
Good point! Didn't think of that. Do you know how to solve that issue? `$this->isObjectType($node, new ObjectType('Illuminate\Routing\ResponseFactory'))` is `false` in that case...

Reply:
The `response()` function returns an instance of `Illuminate\Contracts\Routing\ResponseFactory`, but it may be overkill to check the object type 😅 . It might suffice to just check that it's a FuncCall, with no arguments.

```php
if (! $node->var instanceof FuncCall) {
    return null;
}

if (count($node->var->args) > 0) {
    return null;
}
```

I'm curious, when/why would you use this rule?

Reply:
If you're going the ObjectType way, you'd have to use the `$node->var` in the check, like this `$this->isObjectType($node->var, new ObjectType('Illuminate\Routing\ResponseFactory'))`

Reply:
@GeniJaho Thanks for your input, this fixed the issues!

 The reason I wrote this Rector is that I like this style better than using all the helper functions laravel provides to keep the code more "OOP Style" if you will. It's easier to follow what really happens at first glance.